src/server/internalServer.cpp

```diff
@@ -744,6 +747,41 @@ std::unique_ptr<Response> InternalServer::handle_viewer_settings(const RequestCo
   return ContentResponse::build(*this, RESOURCE::templates::viewer_settings_js, data, "application/javascript; charset=utf-8");
 }
 
+std::unique_ptr<Response> InternalServer::handle_no_js(const RequestContext& request)
+{
+
+  const auto url = request.get_url();
+  const auto domains = kiwix::split(url, "/", true, false);
+  auto filter = get_search_filter(request);
+  try {
+    if (request.get_argument("category") == "") {
+      filter.clearCategory();
+    }
+    if (request.get_argument("lang") == "") {
+      filter.clearLang();
+    }
+  } catch (...) {}
```

Comment:
I'm surprised about that. In which context `get_search_filter` will set a filter on the category and we need to clear it ?
Shouldn't be better to fix `get_search_filter` if it is buggy ?

Reply:
My reasoning for not touching `get_search_filter` was if the behaviour was intentional to display no books when lang/category == "" and if I change that logic, I would break some other functionality.
In kiwix-serve welcome page filters, "" means all categories while `get_search_filter` seems to consider "" as a valid category and rightly shows no book associated

Reply:
I had assumed `category==""` was for all categories filter
Is there a specific category for "all categories" then?

Reply:
I'm not sure about what we are speaking about.
But if we are speaking about the request send to the http api, if we don't want to filter on the category, we must not pass a `category==""` query string.

Reply:
There needs to be a way for the server to know that the user requests "ALL CATEGORIES", I dont think a query param can be cleared without javascript on the client side hence I consider `category=""` as ALL categories & clear that on the server side.

Reply:
> There needs to be a way for the server to know that the user requests "ALL CATEGORIES"

The absence of filter on category seems to me a good way to specify all categories.

> I dont think a query param can be cleared without javascript on the client side hence I consider category="" as ALL categories & clear that on the server side.

Why we need to clear it ?
The html code is generated server side, so we can generate the request we want, without a `category` parameter.

Reply:
> > I dont think a query param can be cleared without javascript on the client side hence I consider category="" as ALL categories & clear that on the server side.
> 
> Why we need to clear it ? The html code is generated server side, so we can generate the request we want, without a `category` parameter.

On the server side we generate the HTML for the search form. The form is filled and submitted from the client side. Values are sent for all inputs of the form (unless it is a request generated via JS).

Reply:
Ha yes, I've forget about the search form.
I would say that `get_search_filter` should be the only (and correct) way to create a filter from query string.
If we add a modification on the filter in specific api, it will introduce inconsistency in the API and we can be sure that we will mix things later. 

> while get_search_filter seems to consider "" as a valid category and rightly shows no book associated.

I finally think it is wrong. The only valid use case of searching for no category is to search for book which have no category associated. So it is more something about `undefined` category than <code></code> category.

Maybe we could extend a bit the api to accept to specific values `[all]` and `[empty]` (or `[none]`) for no filtering (all) and filtering for book without category (empty/none).

The `[]` would prevent a potential conflict with a real category `all` or `none`. 
What do you think about that ?

Reply:
Extending the API would make life easier definitely & bring more consistency.
Would've been nice if `all`/`none` be reserved words for filters but I don't really mind the `[all]` or `[empty/none]` (`none` is better word imo).
Only consideration would be if this looks pleasant on the eye in URL: `TLD/?lang=[all]&category=[none]`